src/Features/Core/Portable/QuickInfo/QuickInfoItem.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    internal sealed class QuickInfoItem
+    {
+        /// <summary>
+        /// The span of the document that the item is based on.
+        /// </summary>
+        public TextSpan Span { get; set; }
+
+        /// <summary>
+        /// Descriptive tags from the <see cref="Microsoft.CodeAnalysis.Completion.CompletionTags"/> type.
+        /// These tags may influence how the item is displayed.
+        /// </summary>
+        public ImmutableArray<string> Tags { get; }
+
+        /// <summary>
+        /// The primary description of the item.
+        /// </summary>
+        public ImmutableArray<TaggedText> Description { get; }
+
+        /// <summary>
+        /// Text derived from the documentation comments in the item's declaration.
+        /// </summary>
+        public ImmutableArray<TaggedText> DocumentationComments { get; }
+
+        /// <summary>
+        /// Text describing the item's type parameters.
+        /// </summary>
+        public ImmutableArray<TaggedText> TypeParameters { get; }
```

Comment:
It might be worth having groups of tagged text with well-defined names to make it more general.